wave_lang/kernel/ops/wave_ops.py

```diff
@@ -1006,6 +1006,29 @@ def indexing_dims(self) -> list[IndexSymbol]:
         combined_dims += get_custom(self.if_false).indexing_dims
         return list(dict.fromkeys(combined_dims))
 
+    def infer_broadcast_shape(self, cond_type, t_type, f_type):
+        c_shape = getattr(cond_type, "symbolic_shape", ())
+        t_shape = getattr(t_type, "symbolic_shape", ())
+        f_shape = getattr(f_type, "symbolic_shape", ())
+
+        # Check if shapes are broadcastable - they should not be disjoint
+        non_empty_shapes = [s for s in [c_shape, t_shape, f_shape] if s]
+        for i in range(len(non_empty_shapes)):
+            for j in range(i + 1, len(non_empty_shapes)):
+                if set(non_empty_shapes[i]).isdisjoint(set(non_empty_shapes[j])):
+                    raise ValueError(
+                        f"SelectOp requires operand shapes to be broadcastable. "
+                        f"Got condition: {c_shape}, if_true: {t_shape}, if_false: {f_shape}"
+                    )
+
+        # Find the most general shape
+        result_shape = ()
+        for t in [cond_type, t_type, f_type]:
+            if hasattr(t, "symbolic_shape"):
+                if len(t.symbolic_shape) >= len(result_shape):
+                    result_shape = t.symbolic_shape
```

Comment:
Thoughts on simplifying this by using previously defined `c_shape , t_shape, and f_shape`

Reply:
or perhaps even the `non_empty_shapes`, something like:

```python
most_general_shape = max(non_empty_shapes, key=len, default=())
```

Reply:
Thanks, I missed this during my clean-up pass.